apps/shell/app.js

```diff
@@ -55,9 +56,67 @@ app.use(process.env.PASSENGER_BASE_URI || '/', router);
 var server = new http.createServer(app);
 var wss = new WebSocket.Server({ noServer: true });
 
+
+
+whitelist = process.env.SSHHOST_WHITELIST;
+if (whitelist === undefined){
+  whitelist = [];
+} else {
+  whitelist = whitelist.split(':');
+}
+
+var filter = '.yml'; // add regex capability
+var files = fs.readdirSync('/etc/ood/config/clusters.d');
+files.forEach( function (file){
+  if (file.indexOf(filter) >= 0){
+    var fullPath = '/etc/ood/config/clusters.d/' + file;
+    var data = yaml.safeLoad(fs.readFileSync(fullPath)); //data is in json format
+    var errorMessage = 'Cluster config file does not follow correct v2 schema. Please see osc.github.io/ood-documentation.';
+
+    if (data.hasOwnProperty('v2')){
```

Comment:
This block seems to want to determine three things:

1. does the cluster config have a hidden == true property in the cluster config (so we would omit it)
2. is a login host specified for the cluster config
3. the host specified as being default?

You could answer these separately by doing:

```javascript
var hidden = data['v2'] && data['v2']['metadata'] && data['v2']['metadata']['hidden'],
      host =  (data['v2'] && data['v2']['login'] && data['v2']['login']['host']) || null,
      default_host = data['v2'] && data['v2']['login'] && data['v2']['login']['default'];

if(default_host)
  whitelist.unshift(host);
else
  whitelist.push(host);
```

The cluster config validation doesn't need to happen in this app.

Reply:
There is a cool method for Ruby hashes called "dig" and there is a third party implementation https://github.com/joe-re/object-dig/blob/master/src/index.js and https://github.com/joe-re/object-dig

This simplifies below

```js
var dig = require('object-dig');


var hidden = dig(data, 'v2', 'metadata', 'hidden'),
      host =  dig(data, 'v2', 'login', 'host'),
      default_host = dig(data, 'v2', 'login', 'default');

if(default_host)
  whitelist.unshift(host);
else
  whitelist.push(host);
```